Automattic/wp-calypso
# Jetpack Manage: Site dashboard v2 - enables sorting via URL from the table column header

Related to https://github.com/Automattic/jetpack-manage/issues/288

## Proposed Changes

* This PR adds sorting in the same style as with the existing dashboard - via the Sites table column header. The only existing sort capability is via URL which made more sense in the previous dashboard (the sites column only included the site URLs).
* The ideal solution to this original issue is to include more sorting options via the API endpoint itself, then we can utlitize the dataviews sorting options as well. See pfunGA-4t-p2#comment-271 for more context. Since the endpoint for the new A4A dashboard has not been created yet, this is just an interim option to keep parity with the existing dashboard.
* As an aside, this PR also removes the 'field' column from the 'View options' toggle - previously it was allowing the ability to display then hide the status column, but that column should be hidden by default and has nothing visible to show. 

## Testing Instructions

* To test, apply this PR and visit `http://jetpack.cloud.localhost:3000/sites`. Notice the up and down arrow icons appearing near the Site column heading.
* Hover over the 'Site' heading and the pointer will display.
* Click on the 'Site heading and the sorting will change. Note the change in the URL to include the new sort parameters. The correct arrow should be highlighted as well, and the sorting organized correctly.
* Note as well the default is to sort by URL, ascending. It takes two clicks to change to descending, but this is the behaviour in the original dashboard as well. After the change is made, each directional change happens on the next click.

**Screenshots**

Default:
<img width="454" alt="Screenshot 2024-02-22 at 09 01 43" src="https://github.com/Automattic/wp-calypso/assets/16754605/dfd28623-3a07-440f-b69b-78b7dcf71e72">

Descending sort direction selected:
<img width="439" alt="Screenshot 2024-02-22 at 09 02 04" src="https://github.com/Automattic/wp-calypso/assets/16754605/43fd2934-6e5f-4e3c-8377-05fc5448319f">


## Pre-merge Checklist
